CCO-21: Support AWS Security Token Service (4.8 GA) (#31136)
openshift/openshift-docs: modules/cco-ccoctl-configuring.adoc

// Module included in the following assemblies:
//
// * authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

[id="cco-ccoctl-configuring_{context}"]
= Configuring the Cloud Credential Operator utility

To create and manage cloud credentials from outside of the cluster when the Cloud Credential Operator (CCO) is operating in manual mode with STS, extract and prepare the CCO utility (`ccoctl`) binary.

[NOTE]
====

> The `ccoctl` is a Linux binary, and must be run in a Linux environment.

Reviewer: Maybe this sounds better than "be run"?

```suggestion
The `ccoctl` is a Linux binary that must run in a Linux environment.
```

// Module included in the following assemblies:
//
// * authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

[id="cco-ccoctl-configuring_{context}"]
= Configuring the Cloud Credential Operator utility

To create and manage cloud credentials from outside of the cluster when the Cloud Credential Operator (CCO) is operating in manual mode with STS, extract and prepare the CCO utility (`ccoctl`) binary.

.Prerequisites

* Obtain the {product-title} release image.

> // Pulling the ccoctl tool from the release image is supported in Linux only, so can this be done outside Linux?

Author: Probably best to just cover the Linux case entirely then, to avoid extra if/then complications.

.Procedure

. Set the `CCO_IMAGE` environment variable:
// Is that what we're doing?
+
[source,terminal]
----
$ CCO_IMAGE=$(oc adm release info --image-for='cloud-credential-operator' $RELEASE_IMAGE)
----

. Extract `ccoctl` from the release image:

> // Is that what we're doing?

Author: ok, attempted to clarify a bit

+
[source,terminal]
----
$ oc image extract $CCO_IMAGE --file="/usr/bin/ccoctl" -a ~/.pull-secret
----
+
.Example output:
+
[source,terminal]
----
// Would this actually be useful here?
----

. Change the permissions to make `ccoctl` executable:
+
[source,terminal]
----
$ chmod 775 ccoctl
----

.Verification

* To verify that `ccoctl` is ready to use, display the help file:
+
[source,terminal]
----
$ ccoctl aws --help
----
+
.Output of `ccoctl aws --help`:
+
[source,terminal]
----
Creating/updating/deleting cloud credentials objects for AWS cloud

Usage:
  ccoctl aws [command]

Available Commands:
  create-all               Create all the required credentials objects
  create-iam-roles         Create IAM roles
  create-identity-provider Create IAM identity provider
  create-key-pair          Create a key pair
  delete                   Delete credentials objects

Flags:
  -h, --help   help for aws

Use "ccoctl aws [command] --help" for more information about a command.
----

> // Is there a reason that the help file for this doesn't list --target-dir and --generate-files-only under "Flags"? Does it matter?

Author: Ah - that makes sense. I like how oc has these consistent recursive help trees :) Will update flags throughout.
